lib/RTCSession.js

```diff
@@ -1344,6 +1342,38 @@ module.exports = class RTCSession extends EventEmitter
     return true;
   }
 
+  replaceMediaStream(mediaStream, options)
+  {
+    debug('replaceMediaStream()');
+
+    return new Promise((resolve, reject) =>
+    {
+      this._removeMediaStream(this._localMediaStream);
+      this._addMediaStream(mediaStream);
+      this.renegotiate(options, resolve, reject);
+    });
+  }
+
+
+  _addMediaStream(stream)
+  {
+    stream.getTracks().forEach((track) => this._connection.addTrack(track, stream));
+  }
+
+  _removeMediaStream(stream)
+  {
+    this._connection.getSenders().forEach((sender) =>
+    {
+      stream.getTracks().forEach((track) =>
+      {
+        if (sender.track === track)
+        {
+          this._connection.removeTrack(sender);
```

Comment:
`break` here.

Reply:
why do you need to break?

Reply:
There is one track for sender. If you found it then there's no reason to keep iterating.

Reply:
wait, we are inside a `forEach`

Reply:
yes, it needs to be rewritten to `for()`

Reply:
```js
if (stream.getTracks().includes(sender.track) {
  this._connection.removeTrack(sender);
}
```

Reply:
no need for the nested `forEach`

Reply:
I thought about
```js
    stream.getTracks().forEach((track) => {
      const senderMediaStream = this._connection.getSenders().find((sender) => sender.track === track);

      if (senderMediaStream) {
        this._connection.removeTrack(senderMediaStream);
      }
    });
``` but it's better

Reply:
or 
```js

    this._connection.getSenders()
    .filter((sender) => stream.getTracks().includes(sender.track))
    .forEach((sender) => { 
      this._connection.removeTrack(sender);
    });
```